lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -85,61 +95,73 @@ def authenticate(group_name=None, internal_token_only=False):
         raise OAuthProblem('Missing authorization header')
     ah_split = auth_header.split()
     if len(ah_split) != 2 or ah_split[0].lower() != 'bearer':
-        logger.warning("invalid auth header. expecting 'bearer' and token with space between")
+        logger.warning(
+            "invalid auth header. expecting 'bearer' and token with space between")
         raise OAuthProblem('Invalid request header')
     _id_token = ah_split[1]
     id_info = verify_userauth(_id_token, group_name, internal_token_only)
-
     return id_info
 
 
 # Checks for group membership to perform authorization
 def authorize(group_name=None):
 
     if 'user_token_data' not in g:
-        raise OAuthProblem('Token data not available for authorization. Most likely an authentication error.')
+        raise OAuthProblem(
+            'Token data not available for authorization. Most likely an authentication error.')
     else:
         id_info = g.user_token_data
 
         if group_name is not None:
-            # check if the group_name is list or string
-            if isinstance(group_name, str):
-                # make group name as list
-                group_name = [group_name]
-
-            is_authorize = False
-            for name in group_name:
-                if uiucedu_is_member_of in id_info:
-                    is_member_of = id_info[uiucedu_is_member_of]
-                    if name in is_member_of:
-                        is_authorize = True
-                        break
-
-            if is_authorize is False:
-                logger.warning("User not authorized.")
+            AUTH_ISSUER = os.getenv(
+                'AUTH_ISSUER', '').strip()
+            # So we are to check is a group membership is required.
+            # Get the membership check keys based on issuer
+            if id_info['iss'] == AUTH_ISSUER:
+                is_member_of_key = is_member_of_claim
+            else:
+                is_member_of_key = uiucedu_is_member_of
+
+            if is_member_of_key in id_info:
```

Comment:
For any specific API endpoint, membership in any one of the required LDAP groups is sufficient. As you can see on the left side, the previous code supports that. For backward compatibility, it needs to be reinstated in some form. For example for POST /events, the user just needs to be member of one of the listed AD groups. https://api-dev.rokwire.illinois.edu/docs/?urls.primaryName=Events%20Building%20Block#/Events/post_events